### PR TITLE
question 3 complete

### PR DESCRIPTION
**Pull request by Akin-23**

### CODE REVIEW

File: __tests__/api-test.test.js [hunk 1]
```diff
@@ -26,3 +28,26 @@ describe("/api/topics", () => {
...
+            "description",
+            expect.any(String)
+          );
+          expect(body[endpoint]).toHaveProperty("queries", expect.any(Array));
+          expect(body[endpoint]).toHaveProperty(
+            "exampleResponse",
+            expect.any(Object)
+          );
+        }
+        expect(Object.keys(body).length).toBe(count);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api-test.test.js]: This is possibly a bit "over engineered" , you can simplify a bit by requiring in the `endpoints.json` file, and asserting that your API has responded with an object that looks exactly like that object

File: controllers/endpoints-controller.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { retrieveEndpoints } = require("../models/endpoints-model");
+
+exports.getEndpoints = (req, res, next) => {
+    retrieveEndpoints().then((data) => {
+        const endpoints = JSON.parse(data);
+        res.status(200).send(endpoints);
+    })
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/endpoints-controller.js:7]: Nicely done

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Good stuff, I think you could shorten the test down by requiring in the file to the test suite but this works too! 👍

File: __tests__/api-test.test.js [hunk 3]
```diff
@@ -26,3 +29,14 @@ describe("/api/topics", () => {
...
 });
 
+describe("/api", () => {
+  test("GET:200 sends object describing all the available endpoints in API", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(endpointObj);
+        expect(Object.keys(body).length).toBe(Object.keys(endpointObj).length);
```
Comment (2202Hannah) [inline __tests__/api-test.test.js:39]: Line 39 isn't necessarily adding anything to your test here. The previous line makes sure that the content is what you expect it to be

**Review by 2202Hannah — COMMENTED** [review]
Nice work on this ticket. I have added an in line comment. Once you have looked over this you can merge to main.

**Pull request merged**